src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -60,8 +71,10 @@
       onPress={onChannelPressed}
       unreadCount={item?.unreadCount}
       isMe={isMe}
-      hasFollowButton={hasFollowButton}
-      handleFollow={handleFollow}
+      hasFollowButton={followStatus?.isFollowing === false || followStatus?.isFollowingFromAction}
+      // hasFollowButton={true}
+      showFollowingButton={followStatus?.isFollowingFromAction}
+      handleFollow={onFollowUser}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The properties `hasFollowButton` and `showFollowingButton` are updated based on the `followStatus`. This is a good practice as it makes the UI more responsive to state changes. However, make sure that the `followStatus` object always has the `isFollowing` and `isFollowingFromAction` properties to avoid potential undefined errors.
```diff
- hasFollowButton={hasFollowButton}
- handleFollow={handleFollow}
+ hasFollowButton={followStatus?.isFollowing === false || followStatus?.isFollowingFromAction}
+ showFollowingButton={followStatus?.isFollowingFromAction}
+ handleFollow={onFollowUser}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->